src/Workspaces/Core/Portable/Shared/Extensions/TextSpanExtensions.cs

```diff
@@ -43,5 +43,20 @@ public static TextSpan Collapse(this IEnumerable<TextSpan> spans)
 
             return TextSpan.FromBounds(start, end);
         }
+
+        /// <summary>
+        /// Returns true if the span encompasses the specified node or token and is contained within its trivia.
+        /// </summary>
+        public static bool IsAround(this TextSpan span, SyntaxNodeOrToken node) => IsAround(span, node, node);
+
+        /// <summary>
+        /// Returns true if the span encompasses a span between the specified nodes or tokens
+        /// and is contained within trivia around them.
+        /// </summary>
+        public static bool IsAround(this TextSpan span, SyntaxNodeOrToken startNode, SyntaxNodeOrToken endNode)
+        {
+            return span.Contains(TextSpan.FromBounds(startNode.Span.Start, endNode.Span.End)) &&
+                   TextSpan.FromBounds(startNode.FullSpan.Start, endNode.FullSpan.End).Contains(span);
```

Comment:
in terms of readability, you might consider:

```c#
var innerSpan = TextSpan.FromBounds(startNode.Span.Start, endNode.Span.End);
var outerSpan = TextSpan.FromBounds(startNode.FullSpan.Start, endNode.FullSpan.End);
return outerSpan.Contains(span) && span.Contains(innerSpan);
```
            return span.Contains(TextSpan.FromBounds(startNode.Span.Start, endNode.Span.End)) &&
                   TextSpan.FromBounds(startNode.FullSpan.Start, endNode.FullSpan.End).Contains(span);

Reply:
➡️ Fixed in 21720afa99f1bc8346e424f59ad97be5315a8a60